controllers/common/predicate.go

```diff
@@ -195,7 +196,7 @@ func ClusterManagerChangedPredicate() predicate.Predicate {
 	err := predicate.Funcs{
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			// This update is in fact a Delete event, process it
-			if _, ok := e.ObjectNew.(*enterpriseApi.ClusterMaster); !ok {
+			if _, ok := e.ObjectNew.(*enterpriseApiV3.ClusterMaster); !ok {
```

Comment:
@vivekr-splunk @mgaldino-splunk shouldn't this function name be `ClusterMasterChangedPredicate` since it is referring to ClusterMaster?

Reply:
I think thats ok, we will reuse the same function for clustermanager also.

Reply:
Because this is attached to IndexerCluster, and we have it in both V3 and V4 I duplicated it in the latest commit. Then it shouldn't be any conflict.